CommonTools/PileupAlgos/plugins/PuppiProducer.cc

```diff
@@ -160,9 +158,9 @@ void PuppiProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
             pReco.id = 1;
           } else if (tmpFromPV == 1 || tmpFromPV == 2) {
             pReco.id = 0;
-            if ((fPtMaxCharged > 0) and (pReco.pt > fPtMaxCharged))
+            if (pReco.pt > fPtMaxCharged)
```

Comment:
I think this should be reverted. As such it will not reproduce the previous behavior in case of (pathologically) large pt's. And in any case, as such it creates an asymmetry wrt how ```PtMaxPhotons``` is dealt with.
My suggestion was simply to assign a value "-1" instead of "0" to the  ```PtMaxCharged``` and  ```PtMaxPhotons``` parameters, so that the condition ```(fPtMaxCharged > 0)``` is always satisfied, even in the unlikely case of floating point upward fluctuations of "0".